scRNA-seq/01-normalizing_scRNA-seq.Rmd

```diff
@@ -61,197 +68,367 @@ and could use these same steps to filter and normalize the data.
+
+### Number of samples that express a gene as a quality measure
+
+Now we have an idea of what samples we probably want to get rid of.
+But what if our data contains genes that we can't reliably measure in these cells?
+Some gene types, like many long non-coding RNAs for example, are very tissue 
+specific and wouldn't be useful for us to look at it. 
+For example if only one cell is supposedly expressing a particular gene at a low
+level, do we think that is biologically relevant, or is it more likely 
+_gene dropout_, due to some amplification issue? 
+Because our single-cell data is already riddled with zeroes, it's best just to 
+get rid of these genes that we suspect are not being reliably measured by many 
+cells. 
+To do this, we will use our detection matrix to add up how many samples express
+each of these genes. 
+
+```{r}
+# Make a vector that contains the number of cells that express a particular gene
+num_samples_that_exp <- apply(detection_mat, 1, sum)
+
+# Let's take a look at what this looks like:
+qplot(num_samples_that_exp, geom = "density") + 
+  geom_density(fill = "lightblue") +
+  geom_vline(xintercept = 10, col = "red") +
+  cowplot::theme_cowplot() 
 ```
 
-Let's plot the relationship of gene variance and gene means so we can get a 
-better idea of what kinds of genes we are working with.
-Here we will plot the `log(variance/mean)` against the `log(mean)` expression 
-for each gene.
-On your own, it's good practice to see how this graph changes 
-with changes of normalization parameters, as well as high variance gene cutoffs.
-
-```{r Plot Variance vs Means}
-# Extract variance stats from Seurat object
-var_stats <- data.frame(seurat_norm@hvg.info)
-
-# Let's plot this with ggplot2
-ggplot(var_stats, aes(x = gene.mean, y = gene.dispersion.scaled)) +
-  # We are using alpha so that the points are more see-through
-  geom_point(alpha = 0.15) +
-  xlab("Gene log(Mean)") +
-  ylab("Gene log(Variance/Mean)") + 
-  theme_cowplot() 
+### Apply a filter based on our criteria
+
+Now that we've explored our data, let's used the information we've learned to 
+actually filter our data.
+
+```{r Apply filter}
+sc_data_filtered <- GeneMatrixFilter(sc_data_filtered, 
+                                     min_counts = 0, 
+                                     num_samples = 10, 
+                                     num_genes = 750)
+
+# Filter metadata accordingly again
+sc_metadata_filtered <- sc_metadata_filtered %>% 
+  dplyr::filter(geo_accession %in% colnames(sc_data_filtered))
 ```
 
-Because our x axis has the same data that we plotted in the above section, 
-you'll notice that we continue to see zero inflation that our single-cell is
-known for.
-Question: Which the genes plotted here might be the most informative for our further 
-analyses?
+Let's see how our filter did in regards to getting rid of samples that are 
+likely bad quality according to percent mapped reads data.
+
+```{r Check percent mapped reads}
```

Comment:
I like it as a support for our chosen methods, but I can get rid of it if you don't think it adds anything to the conversation.

Reply:
I would take it out because it is a bit of an aside in a notebook that is already on the long side and because I think what this number "should look like" is likely still being worked out (if you have a reference that suggests otherwise please let me know

Reply:
Alrighty. I'll take it out.